app/models/bag.rb

```diff
@@ -21,7 +21,9 @@ def src_path
   end
   
   def dest_path
-    File.join(Rails.application.config.upload['storage_path'],bag_id)
+    prefixes = bag_id.match(/^(..)(..)(..).*/)
+    raise RuntimeError, "bag_id too short" unless prefixes
```

Comment:
Is there a reason this line of bag_id validation should be here rather than at bag creation? If this is the place for it, maybe the error message could mention that it's specifically too short to create a full pairtree path?

Reply:
My thoughts were that it isn't really usual for ActiveModel constructors to throw exceptions (rather things just aren't `valid?`) and that it's possible that `dest_path` could be called without the bag having been saved yet or without `valid?` having been checked. We could just call `valid?` here, but `valid?` is also checking things unrelated to the destination path, so it didn't seem quite right.

(Also, this isn't a full pairtree path - in general we're expecting these IDs to be UUIDs, which would generate a deeper pairtree than is really necessary. A 256^3 fanout should be more than adequate for a long, long time)

Reply:
That makes sense to me; my main feeling was that I wouldn't expect an exception when calling `dest_path`. My only suggestion would be to have some sort of default character for truncated ids, but that adds problems like what to do when the default characters are explicitly used, and, if we're expecting UUIDs, then we're also expecting ids to always be at least six characters long.

So yeah, sounds good. ✅